extensions/amp-base-carousel/0.1/test-e2e/test-multi-visible-single-advance.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getNextArrow, getPrevArrow, getSlides} from './helpers';
+import sleep from 'sleep-promise';
+
+describes.endtoend(
+  'AMP carousel advancing with multiple visible',
+  {
+    testUrl:
+      'http://localhost:8000/test/fixtures/e2e/amp-carousel/0.1/multi-visible-single-advance.amp.html',
+    experiments: ['amp-base-carousel'],
+    environments: ['single'],
+  },
+  async function (env) {
+    let controller;
+    let prevArrow;
+    let nextArrow;
+
+    function rect(el) {
+      return controller.getElementRect(el);
+    }
+
+    beforeEach(async () => {
+      controller = env.controller;
+      nextArrow = await getNextArrow(controller);
+      prevArrow = await getPrevArrow(controller);
+    });
+
+    it('should not go forward past end and it should be able to go back correctly', async () => {
+      const slideCount = 7;
+      const slidesInView = 3;
+      const slides = await getSlides(controller);
+
+      // Click `next` more than necessary
+      for (let i = 0; i < 7; i++) {
+        await controller.click(nextArrow);
+        // Need to sleep due to amp-base-carousel buffering clicks
+        await sleep(1000);
```

Comment:
super nit: how'd you decide on the number? does it also work with 500?

Reply:
This caused a problem earlier with the test, so I'd rather err on the side of a little longer if it means not flaky down the road.